### PR TITLE
Add corePKCS11 snippet to README.md

### PR DESCRIPTION
**Pull request by lundinc2**
*Issue #, if available:*

*Description of changes:*
Adds a snippet to the README for the corePKCS11 library.

By submitting this pull request, I confirm that my contribution is made under the terms of the MIT license.

### CODE REVIEW

**Review by sarenameas — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -75,6 +76,12 @@ The [coreJSON](https://github.com/FreeRTOS/coreJSON) library is a JSON parser th
 
 See memory requirements for the latest release [here](https://docs.aws.amazon.com/embedded-csdk/202011.00/lib-ref/libraries/standard/coreJSON/docs/doxygen/output/html/index.html#json_memory_requirements).
 
+#### corePKCS11
+
+The [corePKCS11](https://github.com/FreeRTOS/corePKCS11) library is a software based implementation of the PKCS #11 interface (API) to enable rapid development and flexibility when developing applications that rely on cryptographic operations. Only a subset of [PKCS #11 v2.4](http://docs.oasis-open.org/pkcs11/pkcs11-base/v2.40/os/pkcs11-base-v2.40-os.html) has been implemented, with a focus on operations involving asymmetric keys, random number generation, and hashing.
```
Comment (tanmoysen) [inline README.md:81]: The [corePKCS11] library is an implementation of the PKCS #11 interface (API) that makes it easier to develop applications that rely on cryptographic operations.

File: README.md [hunk 2]
```diff
@@ -75,6 +76,12 @@ The [coreJSON](https://github.com/FreeRTOS/coreJSON) library is a JSON parser th
...
+
+See memory requirements for the latest release [here](https://docs.aws.amazon.com/embedded-csdk/202011.00/lib-ref/libraries/standard/corePKCS11/docs/doxygen/output/html/pkcs11_design.html#pkcs11_memory_requirements).
```
Comment (jainamt) [inline README.md:83]: this memory requirement statement should be added to corepkcs lib's readme, not here. Please see coremqtt section here as well as https://github.com/FreeRTOS/coreMQTT/blob/main/README.md

Comment (jainamt) [inline README.md:83]: also, the link is a dead link.

Comment (abhidixi11) [reply README.md:83]: We do have memory requirement link for other libraries in hub repository.

Comment (lundinc2) [reply README.md:83]: This is a deadlink on purpose, so we don't miss updating it during the release

Comment (jainamt) [reply README.md:83]: Yes, you are right. Sorry, I misspoke.

Comment (jainamt) [reply README.md:83]: Can you please make sure to add this info regarding memory estimate to the corePKCS readme as well? pls see coremqtt readme for example.

Comment (lundinc2) [reply README.md:83]: I have added a PR to add to the README in corePKCS11 https://github.com/FreeRTOS/corePKCS11/pull/60

**Review by jainamt — DISMISSED** [review]

File: README.md [hunk 3]
```diff
@@ -75,6 +76,12 @@ The [coreJSON](https://github.com/FreeRTOS/coreJSON) library is a JSON parser th
 
 See memory requirements for the latest release [here](https://docs.aws.amazon.com/embedded-csdk/202011.00/lib-ref/libraries/standard/coreJSON/docs/doxygen/output/html/index.html#json_memory_requirements).
 
+#### corePKCS11
+
+The [corePKCS11](https://github.com/FreeRTOS/corePKCS11) library is an implementation of the PKCS #11 interface (API) that makes it easier to develop applications that rely on cryptographic operations. Only a subset of [PKCS #11 v2.4](http://docs.oasis-open.org/pkcs11/pkcs11-base/v2.40/os/pkcs11-base-v2.40-os.html) has been implemented, with a focus on operations involving asymmetric keys, random number generation, and hashing.
```
Comment (aggarw13) [inline README.md:81]:
```suggestion
The [corePKCS11](https://github.com/FreeRTOS/corePKCS11) library is an implementation of the PKCS #11 interface (API) that makes it easier to develop applications that rely on cryptographic operations. Only a subset of the [PKCS #11 v2.4](http://docs.oasis-open.org/pkcs11/pkcs11-base/v2.40/os/pkcs11-base-v2.40-os.html) APIs has been implemented, with a focus on operations involving asymmetric keys, random number generation, and hashing.
```

**Review by aggarw13 — APPROVED** [review]

**Review by jainamt — APPROVED** [review]

**Pull request merged**